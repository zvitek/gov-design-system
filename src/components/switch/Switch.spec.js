import { shallowMount } from '@vue/test-utils'
import GovSwitch from '@components/switch/Switch'

let wrapper

describe('GovSwitch', () => {
    beforeEach(() => {
        wrapper = shallowMount(GovSwitch, {
            slots: {
                default: [
                    'Control label'
                ]
            }
        })
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('GovSwitch')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('updates internal value when v-model is changed', () => {
        const newValue = 'switch value'
        wrapper.setProps({ value: newValue })
        expect(wrapper.vm.newValue).toBe(newValue)
    })

    it('emit input event when computedValue is set', () => {
        const newValue = 'switch value'
        wrapper.vm.computedValue = newValue
        const valueEmitted = wrapper.emitted()['input'][0]
        expect(wrapper.vm.newValue).toBe(newValue)
        expect(valueEmitted).toContainEqual(newValue)
    })

    it('method focus() gives focus to the input element', (done) => {
        wrapper.vm.$refs.input.focus = jest.fn()
        wrapper.vm.focus()
        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$refs.input.focus).toHaveBeenCalled()
            done()
        })
    })
})
