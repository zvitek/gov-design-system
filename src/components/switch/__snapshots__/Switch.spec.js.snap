// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovSwitch render correctly 1`] = `<div class="gov-form-control gov-form-control--custom"><input type="checkbox" true-value="true" class="gov-form-control__toggle" value="false"> <label class="gov-form-control__label">Control label</label> <span class="gov-form-control__indicator"></span></div>`;
